libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt

```diff
@@ -0,0 +1,768 @@
+            directionsBuilder
+                    .steps(true)
+                    .geometries(DirectionsCriteria.GEOMETRY_POLYLINE6)
+                    .overview(DirectionsCriteria.OVERVIEW_FULL)
+                    .voiceInstructions(true)
+                    .bannerInstructions(true)
+                    .roundaboutExits(true)
+                    .eventListener(eventListener)
+                    .enableRefresh(true)
+            return NavigationRoute(directionsBuilder.build())
+        }
+
+        private fun parseWaypointIndices(waypointIndices: String): Array<Int> {
+            val splitWaypointIndices = waypointIndices.split(SEMICOLON.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+            val indices = Array(splitWaypointIndices.size, { 0 })
+            var index = 0
+            for (waypointIndex in splitWaypointIndices) {
+                val parsedIndex = Integer.valueOf(waypointIndex)
+                indices[index++] = parsedIndex
+            }
+            return indices
+        }
+
+        private fun parseWaypointTargets(waypointTargets: String): Array<Point?> {
+            val splitWaypointTargets = waypointTargets.split(SEMICOLON.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+            val waypoints = arrayOfNulls<Point>(splitWaypointTargets.size)
+            var index = 0
+            for (waypointTarget in splitWaypointTargets) {
+                val point = waypointTarget.split(COMMA.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+                if (waypointTarget.isEmpty()) {
+                    waypoints[index++] = null
+                } else {
+                    val longitude = java.lang.Double.valueOf(point[0])
+                    val latitude = java.lang.Double.valueOf(point[0])
+                    waypoints[index++] = Point.fromLngLat(longitude, latitude)
+                }
+            }
+            return waypoints
+        }
+
+        private fun assembleWaypoints() {
+            origin?.let { origin ->
+                directionsBuilder.origin(origin.waypoint)
+                directionsBuilder.addBearing(origin.bearingAngle, origin.tolerance)
+            }
+
+            for (waypoint in waypoints) {
+                directionsBuilder.addWaypoint(waypoint.waypoint)
+                directionsBuilder.addBearing(waypoint.bearingAngle, waypoint.tolerance)
+            }
+
+            destination?.let { destination ->
+                directionsBuilder.destination(destination.waypoint)
+                directionsBuilder.addBearing(destination.bearingAngle, destination.tolerance)
+            }
+        }
+    }
+
+    companion object {
+
+        /**
+         * Build a new [NavigationRoute] object with the proper navigation parameters already setup.
+         *
+         * @return a [Builder] object for creating this object
+         * @since 0.5.0
+         */
+        @JvmStatic
+        fun builder(context: Context): Builder {
```

Comment:
Could we make smth like this?
```suggestion
        fun builder(context: Context): Builder =
           builder(context, LocaleUtils())
```